### PR TITLE
Feature/ADF-207/Add new item management roles and assign permissions to them

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-207](https://oat-sa.atlassian.net/browse/ADF-207)_

**Changes:**
* added `Item Class Schema Manager` role;
* added `Item Viewer` role;
* added `Item Manager` role;
* added `Item Previewer` role;
* added `Item Properties Editor` role.

**Require PR:**
* [oat-sa/tao-core#2977](https://github.com/oat-sa/tao-core/pull/2977)

**Companion PR:**
* [oat-sa/extension-tao-testqti-previewer#137](https://github.com/oat-sa/extension-tao-testqti-previewer/pull/137)

### CODE REVIEW

File: actions/class.Items.php [hunk 1]
```diff
@@ -172,11 +172,20 @@ public function editItem()
...
+
+            $formContainer = new SignedFormInstance(
+                $itemClass,
+                $item,
+                [
+                    FormContainer::CSRF_PROTECTION_OPTION => true,
+                    FormContainer::IS_DISABLED => !$hasWriteAccess,
+                ]
+            );
             $myForm = $formContainer->getForm();
```
Comment (gabrielfs7) [inline actions/class.Items.php:186]: If I am not mistaken, here you already persisted the data, not in the `isSubmited`, so basically it is saving without permissions.

File: models/classes/user/TaoItemsRoles.php [hunk 2]
```diff
@@ -26,8 +26,15 @@ interface TaoItemsRoles
 {
     public const ITEM_AUTHOR = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemAuthor';
     public const ITEM_AUTHOR_ABSTRACT = 'http://www.tao.lu/Ontologies/TAOItem.rdf#AbstractItemAuthor';
-    public const ITEM_MANAGER = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemsManagerRole';
+    public const ITEMS_MANAGER = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemsManagerRole';
```
Comment (gabrielfs7) [inline models/classes/user/TaoItemsRoles.php:29]: Why create a breaking change?

File: manifest.php [hunk 3]
```diff
@@ -63,52 +63,97 @@
...
+        ],
+        [
+            AccessRule::GRANT,
+            TaoItemsRoles::ITEM_CLASS_SCHEMA_MANAGER,
+            ['ext' => 'taoItems', 'mod' => 'Items', 'act' => 'removeClassProperty'],
+        ],
+        [
+            AccessRule::GRANT,
+            TaoItemsRoles::ITEM_CLASS_SCHEMA_MANAGER,
+            ['ext' => 'taoItems', 'mod' => 'Category', 'act' => 'getExposedsByClass'],
```
Comment (gabrielfs7) [inline manifest.php:131]: Are we sure those endpoints are only used here?

File: actions/class.Items.php (hunk 1 above)
Comment (gabrielfs7) [reply actions/class.Items.php:186]: Checked and it is not happening. Thanks @shpran

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/user/TaoItemsRoles.php (hunk 2 above)
Comment (gabrielfs7) [reply models/classes/user/TaoItemsRoles.php:29]: Do not forget to rollback the constant name.

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: composer.json [hunk 4]
```diff
@@ -67,7 +67,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis": ">=14.0.0",
-        "oat-sa/tao-core": ">=48.5.0",
+        "oat-sa/tao-core": ">=48.14.5",
```
Comment (gabrielfs7) [inline composer.json:70]: - [ ] Validate the version before merge

**Review by gabrielfs7 — APPROVED** [review]

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-shpran.playground.kitchen.it.taocloud.org:41481/

**Pull request merged**